gcp_variant_transforms/transforms/partition_variants.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A PTransform for partitioning variants based on their chromosome."""
+
+from __future__ import absolute_import
+
+import apache_beam as beam
+import mmh3
+import re
+import collections
+from collections import defaultdict
+
+__all__ = ['PartitionVariants']
+
+# A reg exp that will match to names such as "ch12", "Chr22", or simply "13"
+_CHROMOSOME_NAME_REGEXP = re.compile(r'^(c|ch|chr)?([0-9][0-9]?)$')
+# Partition 0 to 21 is reserved for chromosomes with a common name, ie "ch1".
+_RESERVED_PARTITIONS = 22
+
+class PartitionVariants(beam.PartitionFn):
+  """Partitions variants based on thier chromosome."""
+
+  def __init__(self):
+    """Initializes the transform, mainly a dic of chrom->index for quick look up."""
+
+    self._by_chr = collections.defaultdict(int)
```

Comment:
It doesnt' look like you're actually using the defaultdict functionality, so why not just use a simple dict?